src/ble/ballot_leader_election.rs

```diff
@@ -0,0 +1,224 @@
+use crate::ble::messages::{BleMessage, HeartbeatMsg, HeartbeatReply, HeartbeatRequest};
+use crate::leader_election::{Leader, Round};
+
+/// Used to define an epoch
+#[derive(Clone, Copy, Eq, Debug, Default, Ord, PartialOrd, PartialEq)]
+pub struct Ballot {
+    /// Ballot number
+    pub n: u32,
+    /// The pid of the process
+    pub pid: u64,
+}
+
+impl Ballot {
+    /// Create a new Ballot
+    pub fn with(n: u32, pid: u64) -> Ballot {
+        Ballot { n, pid }
+    }
+}
+
+impl Round for Ballot {}
+
+/// A Ballot Leader Election component. Used in conjunction with Omni-Paxos handles the election of a leader for a group of omni-paxos replicas,
+/// incoming messages and produces outgoing messages that the user has to fetch periodically and send using a network implementation.
+/// User also has to periodically fetch the decided entries that are guaranteed to be strongly consistent and linearizable, and therefore also safe to be used in the higher level application.
+pub struct BallotLeaderComp {
+    pid: u64,
+    pub(crate) peers: Vec<u64>,
+    hb_round: u32,
+    ballots: Vec<(Ballot, bool)>,
+    current_ballot: Ballot, // (round, pid)
+    candidate: bool,
+    leader: Option<Ballot>,
```

Comment:
I think the user should be able to read who is the current leader at any time. So let's create a public function `get_leader()` that returns this.